playground/DatabaseMigration/DatabaseMigration.MigrationService/ApiDbInitializer.cs

```diff
@@ -18,19 +20,29 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
     {
         using var scope = serviceProvider.CreateScope();
         var dbContext = scope.ServiceProvider.GetRequiredService<MyDb1Context>();
+        var dbCreator = dbContext.GetService<IRelationalDatabaseCreator>();
 
-        await InitializeDatabaseAsync(dbContext, cancellationToken);
+        await InitializeDatabaseAsync(dbContext, dbCreator, cancellationToken);
 
         hostApplicationLifetime.StopApplication();
     }
 
-    private static async Task InitializeDatabaseAsync(MyDb1Context dbContext, CancellationToken cancellationToken)
+    private static async Task InitializeDatabaseAsync(MyDb1Context dbContext, IRelationalDatabaseCreator dbCreator, CancellationToken cancellationToken)
     {
         using var activity = s_activitySource.StartActivity("Migrating database", ActivityKind.Client);
 
+        // Database container might not be ready yet, so we need to retry.
         var strategy = dbContext.Database.CreateExecutionStrategy();
         await strategy.ExecuteAsync(async () =>
         {
+            // Create the database if it does not exist.
+            // Do this first so there is then a database to start a transaction against.
+            if (!await dbCreator.ExistsAsync(cancellationToken))
+            {
+                await dbCreator.CreateAsync(cancellationToken);
+            }
```

Comment:
I'd move these to a separate `.ExecuteAsync` call, since they could use up all retries while waiting for the service to start and if there's any delay creating the database `BeginTransactionAsync` will just fail without retrying.